### PR TITLE
fix(site): disable auto fields when they are disabled in the template settings

### LINKED ISSUE #9820: bug: disabling auto-start still lets the user think they can set auto-start
If a template admin disable workspace auto-start, a workspace user still is able to toggle auto-start and click submit.

No error message is shown.

Expected behavior is the auto-start section is disabled for the user.

### PR DESCRIPTION
**Pull request by BrunoQuaresma**
- Disable form inputs 
- Add disable badge + tooltip with more info

Screen Shot 2023-10-03 at 14 20 26

Fix https://github.com/coder/coder/issues/9820

### CODE REVIEW

File: site/src/components/Pill/Pill.tsx [hunk 1]
```diff
@@ -44,61 +44,71 @@ const themeStyles =
...
-            line-height: 0;
-            display: flex;
-            align-items: center;
-            justify-content: center;
-
-            & > img,
-            & > svg {
+    return (
+      <div
+        ref={ref}
```
Comment (code-asher) [inline site/src/components/Pill/Pill.tsx:67]: I am probably missing something obvious, but what does passing in `ref` here do?

**Review by code-asher — APPROVED** [review]
Looks good!  One thing I noticed is that if you already had auto-start/stop set up and then they are disabled via the template, you can still change the time/day/hour fields and you can still hit `Submit` but nothing actually updates.  Should we disable those fields as well?

**Comment by code-asher** [comment]
Whoops lol I misread my GitHub notifications and just noticed this was assigned to someone else, I will remove the assignment since I already reviewed haha.  Sorry all.

File: site/src/components/Pill/Pill.tsx [hunk 2]
```diff
@@ -44,61 +44,71 @@ const themeStyles =
...
+    } = props;
+    const theme = useTheme();
 
-  const typeStyles = useMemo(() => {
-    if (type in themeOverrides) {
-      return themeOverrides[type as keyof typeof themeOverrides](lightBorder);
-    }
-    return themeStyles(type, lightBorder);
-  }, [type, lightBorder]);
+    const typeStyles = useMemo(() => {
```
Comment (Parkreiner) [inline site/src/components/Pill/Pill.tsx:58]: Does this need to be memoized? I guess it just feels like the logic wouldn't have much overhead, and it's only being used within the same component

File: site/src/components/Pill/Pill.tsx (hunk 1 above)
Comment (Parkreiner) [reply site/src/components/Pill/Pill.tsx:67]: The ref gives any parent component the ability to manipulate the the div element created by Pill. 

So for example:
```tsx
function Parent () {
  const divRef = useRef<HTMLDivElement>(null);
  return <Pill ref={divRef} {...otherProps} />
}
```
With this setup, the parent would be able to manipulate the div in effects and event handlers, even though it technically exists in a separate component

Comment (Parkreiner) [reply site/src/components/Pill/Pill.tsx:67]: But judging from this PR, I think it's a preemptive change to make the component more reusable. I don't see any of the `<Pill />` instances in the rest of the code using it

**Review by buenos-nachos — APPROVED** [review]
Looks good! Mainly commenting to reply to Asher

File: site/src/components/Pill/Pill.tsx (hunk 1 above)
Comment (BrunoQuaresma) [reply site/src/components/Pill/Pill.tsx:67]: Yes, it is used by the `Tooltip` component that wraps the `Pill`

File: site/src/components/Pill/Pill.tsx (hunk 2 above)
Comment (BrunoQuaresma) [reply site/src/components/Pill/Pill.tsx:58]: I don't think so, I guess it is only using `memo` as a way to wrap/encapsulate it into a function but I'm just guessing.

File: site/src/components/Pill/Pill.tsx (hunk 1 above)
Comment (code-asher) [reply site/src/components/Pill/Pill.tsx:67]: Ohhhhh I see.  I see it in the docs.  Unfortunate that it is not reflected in the code!  I guess you just have to know how MUI tooltip works.  Thanks for the explanations!

**Comment by code-asher** [comment]
Just reposting this because I did not see a reply to it:   

One thing I noticed is that if you already had auto-start/stop set up and then they are disabled via the template, you can still change the time/day/hour fields and you can still hit Submit but nothing actually updates. Should we disable those fields as well?

**Comment by BrunoQuaresma** [comment]
@code-asher I missed this comment, sorry! I think what you suggest is the right move. I will fix it.

**Pull request merged**